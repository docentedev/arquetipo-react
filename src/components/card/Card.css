.card {
    background: white;
    border-radius: 15px;
    margin: 15px;
    padding: 30px;
    width: 200px;
    border: solid 1px #ededed;
    box-shadow: 0px 1px 9px #d0d0d07d;
}