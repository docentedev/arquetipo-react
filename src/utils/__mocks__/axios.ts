import User from "../../types/user";

export const user : User = {
    name: 'Juan',
    email: 'juan@mail.com',
    username: 'juan',
    address: '',
    company: {
        name: 'string',
        catchPhrase: 'string',
        bs: 'string',
    },
    id: 1,
    phone: '123454325',
    website: 'www.juan.cl',

};


const axios = () => {
    return new Promise((resolve) => {
        resolve({ data: [user] });
    });
};

export default axios;