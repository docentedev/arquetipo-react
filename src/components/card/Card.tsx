import React from 'react';

import './Card.css';

import User from '../../types/user';
import texts from '../../config/texts';

/**
 * Card
 * este comp...
 * 
 * @author Claudio Rojas <claudio.dcv@gmail.com>
 * @since v0.1.0
 * @version v1.0.0
 * @param props
 */
const Card: React.FC<{user : User}> = (props) => {
  return (
    <div className="card">
      <strong>#{props.user.id}<br />{texts.name}: {props.user.name}</strong>
      <p>{props.user.email} | {props.user.phone}</p>
    </div>
  );
};
  
export default Card;