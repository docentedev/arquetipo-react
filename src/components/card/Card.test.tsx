import React from 'react';
import Card from './Card';
import renderer from 'react-test-renderer';
import { user } from '../../utils/__mocks__/axios';

test('Card is ok renderer', () => {
    const component = renderer.create(
        <Card user={user} />,
      );
      let tree = component.toJSON();
      expect(tree).toMatchSnapshot();
});