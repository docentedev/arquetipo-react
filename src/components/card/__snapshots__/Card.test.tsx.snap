// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card is ok renderer 1`] = `
<div
  className="card"
>
  <strong>
    #
    1
    <br />
    Nombre
    : 
    Juan
  </strong>
  <p>
    juan@mail.com
     | 
    123454325
  </p>
</div>
`;
